Dockerfile

```diff
@@ -52,9 +59,27 @@ RUN sed -ri 's,(IGB_UIO=).*,\1n,' config/common_linux* && \
     make $MAKEFLAGS EXTRA_CFLAGS="-march=$CPU -g -w -fPIC -DALLOW_EXPERIMENTAL_API"
 
 WORKDIR /
+
+ARG ENABLE_NTF
+ARG NTF_COMMIT=master
+
 ARG BESS_COMMIT=master
 RUN apt-get update && apt-get install -y wget unzip ca-certificates git
 RUN wget -qO bess.zip https://github.com/NetSys/bess/archive/${BESS_COMMIT}.zip && unzip bess.zip
+
+RUN if [ x"$ENABLE_NTF" != "x" ]; then \
+    wget -qO ntf.zip https://github.com/Network-Tokens/ntf/archive/${NTF_COMMIT}.zip \
+    && unzip ntf.zip \
+    && cp ntf-${NTF_COMMIT}/bessctl/conf/* bess-${BESS_COMMIT}/bessctl/conf/ \
+    && cp ntf-${NTF_COMMIT}/modules/* bess-${BESS_COMMIT}/core/modules/ \
+    && cp ntf-${NTF_COMMIT}/protobuf/* bess-${BESS_COMMIT}/protobuf/ \
+    && cp ntf-${NTF_COMMIT}/utils/* bess-${BESS_COMMIT}/core/utils/; \
+    wget -qO cjose.zip https://github.com/cisco/cjose/archive/master.zip \
+    && unzip cjose.zip \
+    && cd cjose-master \
+    && ./configure && make && make install; \
+  fi
```

Comment:
move this into a small script instead? we want to move rest of snippets too #35 

example
https://github.com/omec-project/ngic-rtc/blob/master/Dockerfile
https://github.com/omec-project/ngic-rtc/blob/master/install_builddeps.sh
https://github.com/omec-project/ngic-rtc/blob/master/install_rundeps.sh

Reply:
I've had several slow builds now and I think it is probably a good idea to maybe have a `build-scripts` directory or something and put scripts for the various stages (ie: linux headers, dpdk, bess download/install, 3rd party stuff) because putting it all in one script is really bad for caching.

@krsna1729 What do you think about having a `build-scripts` or `docker-scripts` directory or something, and having a snippet for each sizable component? This way they can be included from the Dockerfile on a single line to help with caching (ie: so we don't rebuild DPDK if a protobuf definition in a BESS plugin changes...!)

Reply:
`scripts/` makes sense,. ideally you should get the caching benefit if the file or directory COPY-ied was untouched. Today i already get caching benefit for everything until bess build. lets start with -

1. everything till/including DPDK
2. NTF
3. bess

we can also start with just NTF if you want.